src/services/params.ts

```diff
@@ -138,9 +105,9 @@ const jsonParam: ParamGetSet<unknown> = {
 
 export function getParamValue<T extends Param>(value: string | undefined, param: T): ExtractParamType<T>
 export function getParamValue<T extends Param>(value: string | undefined, param: T): unknown {
-  if (value === undefined) {
+  if (value === undefined || !stringHasValue(value)) {
```

Comment:
this brings the same functionality as was previously established in the `optional` utility I removed

<img width="529" alt="image" src="https://github.com/kitbagjs/router/assets/6098901/97908e69-04cd-46b2-984d-8d247bc916b9">

Previously, the optional utility wrapped the param and first checked for `!stringHasValue` and then would return `undefined`.  With the new paradigm our logic for what to do with optional params is handled here instead of in that wrapper param.

One difference this introduces is now an empty string will always throw an `InvalidRouteParamValueError` if the param is not optional.

I'm checking both conditions here even though `stringHasValue` also checks `value === undefined`. This is to ensure the type for `value` is narrowed to exclude `undefined` for the conditions below this one.